protobuf_test.go

```diff
@@ -217,3 +216,47 @@ func TestTimeTypesEncodeDecode(t *testing.T) {
 	assert.Equal(t, in.Time.UnixNano(), out.Time.UnixNano())
 	assert.Equal(t, in.Duration, out.Duration)
 }
+
+type cipherText struct {
+	A, B *int32
+}
+
+type testMsg struct {
+	M map[uint32][]cipherText
+}
+
+func TestMapSliceStruct(t *testing.T) {
+	// equivalent to the encoding to the following in a .proto file:
+	// details: https://developers.google.com/protocol-buffers/docs/proto#backwards-compatibility
+	/*
```

Comment:
Can you put that before the method definition, as it's not really relevant to the method itself, but to the structures you define above.